src/openms/source/CHEMISTRY/RibonucleotideDB.cpp

```diff
@@ -66,6 +87,209 @@ namespace OpenMS
     }
   }
 
+  // All valid JSON ribonucleotides must at minimum have elements defining name, short_name, reference_moiety, and formula
+  bool entryIsWellFormed_(const nlohmann::json::value_type& entry)
+  {
+    if (entry.find("name") == entry.cend())
+    {
+      String msg = "\"name\" entry missing for ribonucleotide";
+      throw Exception::MissingInformation(__FILE__, __LINE__,
+                                              OPENMS_PRETTY_FUNCTION, msg);
+    }
+    if (entry.find("short_name") == entry.cend())
+    {
+      String msg = "\"short_name\" entry missing for ribonucleotide";
+      throw Exception::MissingInformation(__FILE__, __LINE__,
+                                              OPENMS_PRETTY_FUNCTION, msg);
+    }
+    if (entry.find("reference_moiety") == entry.cend())
+    {
+      String msg = "\"reference_moiety\" entry missing for ribonucleotide";
+      throw Exception::MissingInformation(__FILE__, __LINE__,
+                                              OPENMS_PRETTY_FUNCTION, msg);
+    }
+    if (entry.find("formula") == entry.cend())
+    {
+      String msg = "\"formula\" entry missing for ribonucleotide";
+      throw Exception::MissingInformation(__FILE__, __LINE__,
+                                              OPENMS_PRETTY_FUNCTION, msg);
+    }
+    return true;
+  }
+  
+  EmpiricalFormula getBaseLossFormula_(const nlohmann::json::value_type& entry)
+  {
+    String code = entry.at("short_name");
+    // If we have an explicitly defined baseloss_formula
+    if (auto e = entry.find("baseloss_formula"); e != entry.cend() && !e->is_null())
+    {
+      return EmpiricalFormula(entry.at("baseloss_formula"));
```

Comment:
```suggestion
      return EmpiricalFormula(*e);
```

don't look it up again. You have it right here.